tests/ci/integration/run_netty_tcnative_integration.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+set -euxo pipefail
+
+source tests/ci/common_posix_setup.sh
+
+# Set up environment.
+
+# SRC_ROOT(aws-lc)
+#  - SCRATCH_FOLDER
+#    - NETTY_TCNATIVE_SRC
+#    - AWS_LC_BUILD_FOLDER
+#    - AWS_LC_INSTALL_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER=${SRC_ROOT}/"scratch"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+NETTY_TCNATIVE_SRC="${SCRATCH_FOLDER}/netty-tcnative"
+NETTY_TCNATIVE_GIT_URL="https://github.com/netty/netty-tcnative.git"
+NETTY_TCNATIVE_REFSPEC="${1:-main}"
+NETTY_TCNATIVE_PATCH_FOLDER="${SRC_ROOT}/tests/ci/integration/netty_tcnative_patch"
+
+function build_netty_tcnative() {
+    pushd "${NETTY_TCNATIVE_SRC}"
+    
+    local PKG_CONFIG_PATH
+    PKG_CONFIG_PATH="$(find "${AWS_LC_INSTALL_FOLDER}" -type d -name pkgconfig)"
+    export PKG_CONFIG_PATH
+    
+    local NETTY_LD_LIBRARY_PATH
+    NETTY_LD_LIBRARY_PATH="$(pkg-config --variable=libdir libcrypto)"
+    
+    local NETTY_LDFLAGS
+    NETTY_LDFLAGS="$(pkg-config --libs libssl libcrypto)"
+    NETTY_LDFLAGS+="${LDFLAGS:-}"
+    
+    local NETTY_CFLAGS
+    NETTY_CFLAGS="$(pkg-config --cflags libssl libcrypto)"
+    NETTY_CFLAGS+="${CFLAGS:-}"
+    
+    env LD_LIBRARY_PATH="${NETTY_LD_LIBRARY_PATH}" LDFLAGS="${NETTY_LDFLAGS}" CFLAGS="${NETTY_CFLAGS}" ./mvnw "$@"
+    
+    unset PKG_CONFIG_PATH
+    popd # "${NETTY_TCNATIVE_SRC}"
+}
+
+
+function clone_and_patch_netty_tcnative() {
+    git clone "${NETTY_TCNATIVE_GIT_URL}" "${NETTY_TCNATIVE_SRC}"
+    pushd "${NETTY_TCNATIVE_SRC}"
+    git fetch origin "${NETTY_TCNATIVE_REFSPEC}"
+    git checkout -b aws-lc "${NETTY_TCNATIVE_REFSPEC}"
+    if [[ -e "${NETTY_TCNATIVE_PATCH_FOLDER}/${NETTY_TCNATIVE_REFSPEC}.patch" ]]; then
+        patch -p1 -i "${NETTY_TCNATIVE_PATCH_FOLDER}/${NETTY_TCNATIVE_REFSPEC}.patch"
+    else
+        patch -p1 --no-backup-if-mismatch -i "${NETTY_TCNATIVE_PATCH_FOLDER}/latest.patch"
+    fi
```

Comment:
Would patches even be needed if we simply set a `-DOPENSSL_IS_BORINGSSL=1` when building?